### ISSUE TITLE
XDP support documentation

### ISSUE THREAD

**Issue by bogdanPricope** [OpenDataPlane/odp#1623]
I don't know if you have this but it will be nice to have a document describing XDP support in ODP

e.g.
1. Linux package needed (and versions) -  this is especially useful for the unfortunate ones that don't have xdp packages in the distributions they use (.e.g Ubuntu).
2. odp configuration options examples
./configure --prefix=<your path> --enable-xdp=yes --with-platform=linux-generic PKG_CONFIG_PATH=/usr/lib64/pkgconfig
4. odp configuration limitations (e.g. cannot have dpdk and xdp in the same time)
5. Env. variables to set for non-standard installations 
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/lib64
export LIBXDP_OBJECT_PATH=/usr/lib64/bpf
7. Packet pool size constraints (1024 x number of queues).
8. Other  constraints and bugs (malformed packets, freeze in odp_pktio_close(), etc.)
9. Performance comparison dpdk vs xdp on some generic use-case

Best regards

**Comment by TuomasTaipale** [comment]
There's some documentation and requirements listed in section 3.6 of `DEPENDENCIES`, though not all you listed here.

Can you elaborate a bit the "... freeze in odp_pktio_close() ..." part? There's a one second delay added to `odp_pktio_close()` on purpose when XDP is in use as cleanup routines with some drivers are asynchronous and can't really handle calling `odp_pktio_open()` quickly after `odp_pktio_close()` without it.

**Comment by bogdanPricope** [comment]
Sure thing, mate..

odp_packet_alloc_multi(1024) returned 1024
odp_packet_alloc_multi(1024) returned 1024
odp_packet_alloc_multi(1024) returned 1023
odp_packet_alloc_multi(1024) returned 0
pktio/socket_xdp.c:948:sock_xdp_output_queues_config():Unable to reserve fill queue descriptors for queue: 10.
E 41 5:4150986752 nfp_ifnet.c:225] Failed to create output queues.
E 41 5:4150986752 nfp_ifnet_portconf.c:84] Error: Failed to create the interface.
Error: failed to create interface ens5f1.
I 41 5:4150986752 nfp_ifnet.c:578] Cleaning device 'ens5f1' addr  02:00:4a:35:a2:fd
odp_packet_io.c:701:_pktio_stop():Not started
^C
Thread 1 "nfp_fpm" received signal SIGINT, Interrupt.
0x00007ffff77481a6 in xsk_umem.delete () from /usr/lib64/libxdp.so.1
(gdb) bt
#0  0x00007ffff77481a6 in xsk_umem.delete () from /usr/lib64/libxdp.so.1
#1  0x00007ffff7e59ae5 in umem_delete (umem_info=<optimized out>) at pktio/socket_xdp.c:191
#2  umem_delete (umem_info=0x7fffe658c900) at pktio/socket_xdp.c:186
#3  sock_xdp_close (pktio_entry=0x7fffe1200040) at pktio/socket_xdp.c:315
#4  0x00007ffff7e1b155 in _pktio_close (entry=entry@entry=0x7fffe1200040) at odp_packet_io.c:433
#5  0x00007ffff7e21658 in odp_pktio_close (hdl=<optimized out>) at odp_packet_io.c:534
#6  0x00005555560609ee in nfp_ifnet_net_cleanup (ifnet=0x7fffe8924200) at nfp_ifnet.c:643
#7  0x000055555606186d in nfp_net_interfaces_destroy () at nfp_ifnet_portconf.c:1783
#8  0x000055555606ddf0 in nfp_initialize (params=<optimized out>, params@entry=0x7fffffffcb10) at nfp_init.c:953
#9  0x000055555580e034 in main (argc=<optimized out>, argv=0x7fffffffe3b8) at app_main.c:111
(gdb) quit
A debugging session is active.

	Inferior 1 [process 182361] will be killed.

Quit anyway? (y or n) y
++++++++++++++++++++++++++++++++++++++++++++++++++

**Comment by TuomasTaipale** [comment]
OK, thanks. What's the kernel version you are using and are you sharing a packet pool between packet I/Os?

**Comment by bogdanPricope** [comment]
# uname -a
Linux DUTX 5.15.0-40-generic #43-Ubuntu SMP Wed Jun 15 12:54:21 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

For that execution I used one pktio and afaik is shared between pktios. odp_packet_alloc_multi() errors was due to lack of packets in the pool (a small pool and many queues)... once I increased to pool size it was fine.... so ignore that part.

**Comment by bogdanPricope** [comment]
btw, in the 'normal' initialization/cleanup procedure, the odp_pktio_close() worked fine... The freeze took place only after sock_xdp_output_queues_config() error ...
